joplin/pages/official_documents_page/models.py

```diff
@@ -88,3 +88,59 @@ class OfficialDocumentPageDocument(Orderable):
 
     class Meta:
         indexes = [models.Index(fields=['-date'])]
+
+
+class OfficialDocumentPage(JanisBasePage):
+    janis_url_page_type = "official_document_page"
+    base_form_class = OfficialDocumentPageForm
+
+    date = models.DateField(verbose_name="Document date", null=True, blank=True)
+    authoring_office = models.CharField(verbose_name="Authoring office of document", max_length=DEFAULT_MAX_LENGTH,
+                                        blank=True)
+    summary = models.TextField(verbose_name="Document summary", blank=True)
+    name = models.CharField(verbose_name="Name of Document", max_length=DEFAULT_MAX_LENGTH, blank=True)
+    document = models.ForeignKey(Document, null=True, blank=True, on_delete=models.SET_NULL, related_name='+',
+                                 verbose_name="Document [en]")
+    document_es = models.ForeignKey(Document, blank=True, null=True, on_delete=models.SET_NULL, related_name='+',
+                                    verbose_name="Document [es]")
+
+    publish_requirements = (
+        FieldPublishRequirement("date",
+                                message="You need to include a date before publishing"),
+        FieldPublishRequirement("authoring_office", langs=["en"],
+                                message="You need to include the authoring office before publishing"),
+        FieldPublishRequirement("summary", langs=["en"],
+                                message="You need to include a summary before publishing"),
+        FieldPublishRequirement("name", langs=["en"],
+                                message="You need to include a name before publishing"),
+        FieldPublishRequirement('document'),
+        RelationPublishRequirement('official_document_collection'),
+    )
+
+    content_panels = [
+        FieldPanel('title_en', widget=countMe),
+        FieldPanel('title_es', widget=countMe),
+        FieldPanel('title_ar'),
+        FieldPanel('title_vi'),
+        FieldPanel('date'),
+        FieldPanel('authoring_office', widget=countMe),
+        FieldPanel('summary', widget=widgets.CountableWidget(attrs={
+            'data-count': 'characters',
+            'data-max-count': AUTHOR_LIMITS['document_summary'],
+            'data-count-direction': 'down'
+        })),
+        FieldPanel('name', widget=countMe),
+        DocumentChooserPanel('document'),
+        DocumentChooserPanel('document_es'),
+        InlinePanel('official_document_collection', label="Official Document Collections this Document belongs to")
+    ]
+
+
+class OfficialDocumentCollectionDocument(ClusterableModel):
```

Comment:
I'd recommend renaming this to a more verbose "OfficialDocumentCollectionOfficialDocumentPage". It's a lot. It's terrible. But that is what it is. Because a "Document" means something else in wagtail but an "OfficialDocumentPage" is what the actual page is called. And the "OfficialDocumentPage" has wagtail "Documents" connected to it. Taking shortcuts on names can create confusion.

Reply:
Duly noted. i will update tomorrow.

Reply:
In that case, the filter should be renamed as well, right?

Reply:
OfficialDocumentCollectionOfficialDocumentPageFilter